src/levelScene.cpp

```diff
@@ -28,9 +29,12 @@ LevelScene::LevelScene()
     disable();
     
     camera_capture_texture = new CameraCaptureTexture();
+
+    //Allocate a big replay buffer so during gameplay we don't have to ask for more to prevent stuttering
+    replay_controls_buffer = std::vector<ControlsState>((size_t)(sp::Engine::fixed_update_frequency * max_replay_time_sec));
```

Comment:
use  replay_controls_buffer.resize(X) instead.

Reply:
I think you meant the reserve method? Changed to that one.

Reply:
Didn't check if you append or put in place. But resize makes the vector a certain size, the reserve only reserves memory for enties but does not resize the actual vector.